GlobalScaling/GlobalScalingTests.c

```diff
@@ -0,0 +1,175 @@
+#include "GlobalScalingTests.h"
+#include "GlobalScaling.h"
+
+void runGlobalScalingTests(){
+    globalScalingCleanTest();
+
+    globalScalingDirtyTest1();
+
+    globalScalingDirtyTest2();
+
+    globalScalingDirtyTest3();
+
+    globalScalingDirtyTest4();
+
+    globalScalingDirtyTest5(); 
+}
+
+/*
+    -First clean test
+    -intializes each index of the matrix to the value of 1
+    -calls globalscaling
+    -checks each index if the correct value was entered at that index
+    -if unsuccessful prints test failed
+*/
+void globalScalingCleanTest(){
+    for(int i = 0; i < 4; i++){
+        for(int j = 0; j < 4; j++){
```

Comment:
for global scaling, isnt it mutiplying by the matrix:
1 0 0 0
0 1 0 0
0 0 1 0
0 0 0 k
where x,y,z are the scale factors for each dimension? This code creates the following matrix:
1 1 1 1
1 1 1 1
1 1 1 1
1 1 1 1

you can remove your nested for loop and just do the following:
```
for( int i = 0; i < 3; i++) {
   transformationMatrix[i][i] = 1;
}
transformationMatrix[3][3] = getGlobalScale();
```